components/table/ReportTable.tsx

```diff
@@ -57,44 +61,94 @@ function TotalHours({
     )
 }
 
-function CustomerHoursRow({
+function ProjectHoursRow({
     entries,
-    customer,
-}: CustomerHoursRowProps): JSX.Element {
+    project,
+}: ProjectHoursRowProps): JSX.Element {
     return (
         <ListItem>
-            Total hours for {customer.name}: {entriesQuantitySum(entries)}
+            Total hours for {project.name}: {entriesQuantitySum(entries)}
         </ListItem>
     )
 }
 
-function ProjectHoursRow({
+function CustomerHoursRow({
     entries,
-    project,
-}: ProjectHoursRowProps): JSX.Element {
+    customer,
+    projects,
+}: CustomerHoursRowProps): JSX.Element {
+    const entriesByProject = (
+        entriesToFilter: TimesheetEntry[],
+        projectId: number
+    ) =>
+        entriesToFilter.filter(
+            (entry) => entry.timesheet.project?.id === projectId
+        )
     return (
-        <ListItem>
-            Total hours for {project.name}: {entriesQuantitySum(entries)}
-        </ListItem>
+        <>
+            <ListItem>
+                Total hours for {customer.name}: {entriesQuantitySum(entries)}
+            </ListItem>
+            {
+                <UnorderedList>
+                    {projects.map(
+                        (project) =>
+                            project.id && (
+                                <ProjectHoursRow
+                                    entries={entriesByProject(
+                                        entries,
+                                        project.id
+                                    )}
+                                    key={`project-hours-row-customer-${project.id}`}
+                                    project={project}
+                                ></ProjectHoursRow>
+                            )
+                    )}
+                </UnorderedList>
+            }
+        </>
     )
 }
 
 function EmployeeHoursRow({
     entries,
     employee,
+    projects,
 }: EmployeeHoursRowProps): JSX.Element {
+    const entriesByProject = (
```

Comment:
The same function is defined on line 80, I'd suggest defining all of these below `entriesQuantitySum` (outside all components):
```typescript
const entriesByCustomer = (
    entriesToFilter: TimesheetEntry[],
    customerId: number
) =>
    entriesToFilter.filter(
        (entry) => entry.timesheet.project.customer?.id === customerId
    )

const entriesByProject = (
    entriesToFilter: TimesheetEntry[],
    projectId: number
) =>
    entriesToFilter.filter((entry) => entry.timesheet.project?.id === projectId)

const entriesByEmployee = (
    entriesToFilter: TimesheetEntry[],
    employeeId: number
) =>
    entriesToFilter.filter(
        (entry) => entry.timesheet.employee?.id === employeeId
    )

const projectByCustomer = (projectsToFilter: Project[], customerId: number) =>
    projectsToFilter.filter((project) => project.customer?.id === customerId)

const timesheetsByEmployee = (
    timesheetsToFilter: Timesheet[],
    employeeId: number
) =>
    timesheetsToFilter
        .filter((timesheet) => timesheet.employee.id === employeeId)
        .map((timesheet) => timesheet.project)
```